src/MessageFormatter.php

```diff
@@ -32,15 +32,19 @@
  * - {res_headers}:    Response headers
  * - {req_body}:       Request body
  * - {res_body}:       Response body
+ *
+ * @final
  */
-class MessageFormatter
+class MessageFormatter implements MessageFormatterInterface
 {
     /**
      * Apache Common Log Format.
      *
      * @link https://httpd.apache.org/docs/2.4/logs.html#common
      *
      * @var string
+     *
+     * @deprecated log format constants will be moved into MessageFormatterInterface in guzzlehttp/guzzle:8.0.
```

Comment:
We can not add a deprecation without providing an alternative.. Yes.. these will be moved to the interface but currently they are not there..

Reply:
Which annotation you suggest?

Reply:
No annotation. Leave things as they were...

Reply:
Wait, why did you wanna move it to the interface, anyway?

Reply:
As this is (Apache) common log format it may be directly in interface. imho.

Reply:
Even if this is a standard format.. you never know why someone might have overwritten it.. it is a BC.. deprecated SHOULD go with an alternative IMHO. Saying to ppl that we deprecate stuff but the alternative is on a FUTURE version is not good.

Reply:
Yep. I please keep the constants on the original class, and don't annotate them with anything. In 8.x we can make the class final, and move the constants to the interface.

Reply:
Annotation removed.